src/services/auth.service.js

```diff
@@ -1,63 +1,83 @@
+    this.model = customer;
+  }
 
-    constructor() {
-        this.model = customer
-    }
-
-    async register(data) {
-        try {
-            let customer
-
-            customer = await this.model.findOne({ where: { email: data.email } })
+  async register(data) {
+    try {
+      let customer;
 
-            if (customer) {
-                throw new AppError('User with entered email already exist', 400)
-            }
+      customer = await this.model.findOne({ where: { email: data.email } });
 
-            customer = await this.model.findOne({ where: { username: data.username } })
+      if (customer) {
+        throw new AppError("User with entered email already exist", 400);
+      }
 
-            if (customer) {
-                throw new AppError('User with entered username already exist', 400)
-            }
+      customer = await this.model.findOne({
+        where: { username: data.username },
+      });
 
-            customer = await this.model.findOne({ where: { bvn: data.bvn } })
+      if (customer) {
+        throw new AppError("User with entered username already exist", 400);
+      }
 
-            if (customer) {
-                throw new AppError('User with entered bvn already exist', 400)
-            }
+      customer = await this.model.findOne({ where: { bvn: data.bvn } });
 
-            customer = await this.model.findOne({ where: { phoneNumber: data.phoneNumber } })
+      if (customer) {
+        throw new AppError("User with entered bvn already exist", 400);
+      }
 
-            if (customer) {
-                throw new AppError('User with entered phone number already exist', 400)
-            }
+      customer = await this.model.findOne({
+        where: { phoneNumber: data.phoneNumber },
+      });
 
-            customer = await this.model.findOne({ where: { nin: data.nin } })
+      if (customer) {
+        throw new AppError("User with entered phone number already exist", 400);
+      }
 
-            if (customer) {
-                throw new AppError('User with entered nin already exist', 400)
-            }
+      customer = await this.model.findOne({ where: { nin: data.nin } });
 
-            const length = await this.model.count()
+      if (customer) {
+        throw new AppError("User with entered nin already exist", 400);
+      }
 
-            const accountNumber = 1000000000 + (length + 1)
+      const length = await this.model.count();
 
-            data.accountNumber = accountNumber
+      const accountNumber = 1000000000 + (length + 1);
 
-            customer = await this.model.create(data)
+      data.accountNumber = accountNumber;
 
-            return customer
+      customer = await this.model.create(data);
 
-        } catch (error) {
-            throw error
-        }
+      return customer;
+    } catch (error) {
+      throw error;
     }
+  }
+  async login(data) {
+    try {
+      const { email, password } = body;
+      let user = await this.model.findOne({
```

Comment:
@EseigbeSamuel can you improve this so we first check for the email and if the email is not found we send a message. After doing that we then check for the password, but remember we created a password match method on customer model that is what we use to check if passwords match. Let me know if you have questions. Great work! Well done